packages/messenger-widget/src/components/ConfigureProfile/MobileView.tsx

```diff
@@ -50,59 +77,245 @@ export function MobileView() {
+                    )}
+                >
+                    {ethAddress &&
+                        ethAddress + dm3Configuration.addressEnsSubdomain}
+                </p>
+                {!isProfileUpdatedForAddrName() && (
+                    <button
+                        className="add-prof-btn-active update-btn"
+                        onClick={() => updateProfileForAddressName()}
+                    >
+                        <img
+                            className="me-1 pointer-cursor "
+                            src={updateIcon}
+                            alt="update profile"
+                        />
+                        update now
+                    </button>
+                )}
             </div>
 
-            {fetchDM3NameComponent(
-                dm3NameServiceSelected,
-                dm3Configuration.chainId,
+            {/* Existing DM3 name */}
+            {existingDm3Name && (
+                <div className="d-flex mt-2 ps-2 flex-column">
+                    <p
+                        className={'m-0 font-size-14 font-weight-500 line-height-24 title-content'.concat(
+                            ' ',
+                            !isProfileUpdatedForDm3Name() ? 'error-text' : '',
+                        )}
+                    >
+                        DM3 Name
+                        <span className="address-tooltip">
+                            i
+                            <span className="address-tooltip-text">
+                                DM3 name is used as a username and can be used
+                                by any address to send the messages to this DM3
+                                name.
+                                <br />
+                                <span className="font-weight-800">
+                                    {' '}
+                                    You can receive messages sent to your DM3
+                                    name.
+                                </span>
+                            </span>
+                        </span>
+                    </p>
+                    <p
+                        className={'dm3-address m-0 ms-0 font-size-14 font-weight-500 line-height-24 grey-text'.concat(
+                            ' ',
+                            !isProfileUpdatedForDm3Name()
+                                ? 'error-text'
+                                : 'grey-text',
+                        )}
+                    >
+                        {existingDm3Name}
+
+                        {/* Delete icon */}
+                        <img
+                            className="ms-4 pointer-cursor del-icon"
+                            src={deleteIcon}
+                            alt="remove"
+                            onClick={() => setShowDeleteConfirmation(true)}
+                        />
+                    </p>
+
+                    {!isProfileUpdatedForDm3Name() && (
```

Comment:
use the useMemo hook to store the results of functions like isProfileUpdatedForDm3Name